public/scripts/main.js

```diff
@@ -0,0 +1,26 @@
+function login() {
+  console.log('youve logged in')
+  fetch('/login', {method: 'post', redirect: 'follow'})
+  .then(response => response.json())
+  .then( response => {
+    if (response.redirectUrl) {
+      window.location = response.redirectUrl
+    } else {
+      console.error('Oh no, error')
+      // Display error
```

Comment:
Try and make the error more explicit, so 'Oh no, Login error' If you don't want to log the actual error.